sys/kern/kern_proc.c

```diff
@@ -2534,8 +2534,8 @@ sysctl_kern_proc_c18n(SYSCTL_HANDLER_ARGS)
 	    info.version != CHERI_C18N_INFO_VERSION ||
 	    info.stats_size == 0 ||
 	    info.stats_size > RTLD_C18N_STATS_MAX_SIZE ||
-	    !__CAP_CHECK(info.stats, info.stats_size) ||
-	    (cheri_getperm(info.stats) & CHERI_PERM_LOAD) == 0) {
+	    !cheri_can_access(info.stats, CHERI_PERM_LOAD,
```

Comment:
These two hunks seem to be unrelated and some other bug fix?

Reply:
This is just replacing the `__CAP_CHECK` pattern with `cheri_can_access` to resemble what `sysctl_kern_proc_c18n_compartments` is doing below.

Reply:
This could be made a separate change, certainly.